### PR TITLE
feat: add permissions package 

### PR DESCRIPTION
**Pull request by jaideepr97**
**What type of PR is this?**
> Uncomment only one ` /kind` line, and delete the rest.
> For example, `> /kind bug` would simply become: `/kind bug`

> /kind bug
> /kind chore
> /kind cleanup
> /kind failing-test
 /kind enhancement
> /kind documentation
> /kind code-refactoring

**What does this PR do / why we need it**:
Adds the permissions package that is responsible for all rbac object generation and CRUD operations on the cluster. Currrently includes roles/rolebindings/clusterroles/clusterrolebindings/serviceaccounts

NOTE: this PR just adds the package files and tests, this code is not invoked at any point yet

NOTE: This PR depends on #947 

**Have you updated the necessary documentation?**

* [ ] Documentation update is required by this PR.
* [ ] Documentation has been updated.

**Which issue(s) this PR fixes**:

Fixes #?

**How to test changes / Special notes to the reviewer**:
Unit tests are included

### CODE REVIEW

File: pkg/argoutil/resource.go [hunk 1]
```diff
@@ -72,6 +72,16 @@ func CreateEvent(client client.Client, eventType, action, message, reason string
 	return client.Create(context.TODO(), event)
 }
 
+// GenerateResourceName generates names for namespace scoped scoped resources
```
Comment (ishitasequeira) [inline pkg/argoutil/resource.go:75]: nit: scoped is written twice

File: pkg/permissions/clusterrole.go [hunk 2]
```diff
@@ -0,0 +1,114 @@
...
+		ObjectMeta: metav1.ObjectMeta{
+			Name:        crName,
+			Labels:      argoutil.MergeMaps(argoutil.LabelsForCluster(instanceName, component), labels),
+			Annotations: argoutil.MergeMaps(argoutil.AnnotationsForCluster(instanceName, instanceNamespace), annotations),
+		},
+		Rules: rules,
+	}
+}
+
+func RequestClusterRole(request ClusterRoleRequest) (*rbacv1.ClusterRole, error) {
```
Comment (ishitasequeira) [inline pkg/permissions/clusterrole.go:46]: Add description to the functions that do not have descriptions

**Review by ishitasequeira — CHANGES_REQUESTED** [review]

File: pkg/permissions/clusterrole.go (hunk 2 above)
Comment (jaideepr97) [reply pkg/permissions/clusterrole.go:49]: done

**Review by iam-veeramalla — APPROVED** [review]
lgtm, thanks @jaideepr97 .. please take a look at @ishitasequeira comments

**Review by ishitasequeira — APPROVED** [review]
LGTM!!

**Pull request merged**